adding component structure chapter (#54)
infinum/eightshift-docs: website/docs/basics/blocks-component-manifest.md

---

[![docs-source](https://img.shields.io/badge/source-eigthshift--frontend--libs-yellow?style=for-the-badge&logo=javascript&labelColor=2a2a2a)](https://github.com/infinum/eightshift-frontend-libs/tree/develop/blocks/init/src/blocks/)

> Component manifest is a bit different than a block manifest. The main difference is that components manifest is a fully custom feature, and it is only here to provide a way to store configuration data and share it with multiple files in your project.

Reviewer: ```suggestion
A component manifest is a bit different than the block manifest. The main difference is that component manifest is a fully custom feature, and it is only here to provide a way to store configuration data and share it with multiple files in your project.
```

### Example

```json
{
  "componentName": "heading",
  "title": "Heading",
  "componentClass": "heading",
  "example": {
    "attributes": {
      "headingContent": "This is heading",
      "headingLevel": 2,
      "headingSize": "default",
      "headingUse": true
    }
  },
  "attributes": {
    "headingContent": {
      "type": "string"
    },
    "headingLevel": {
      "type": "int",
      "default": 2
    },
    "headingSize": {
      "type": "string",
      "default": "default"
    },
    "headingUse": {
      "type": "boolean",
      "default": true
    }
  },
  "options": {
    "sizes": [
      {
        "label": "Default (30px)",
        "value": "default"
      },
      {
        "label": "Big (50px)",
        "value": "big"
      }
    ],
    "aligns": [
      "left",
      "center",
      "right"
    ],
    "colors": [
      "primary",
      "black"
    ]
  },
  "components": {}
}
```

### componentName

This key is not generally used anywhere. We wanted some consistency with the block manifest.

### title

> This key is used to provide the label for all your components options and provide the storybook's automatic story title.

Reviewer: ```suggestion
This key is used to provide the label for all your component options and provide the storybook's automatic story title.
```

### componentClass

> These keys are used to provide a single point of reference for a component name. You use this key to provide a prefix for all css selectors in your component.

Reviewer: ```suggestion
These keys are used to provide a single point of reference for a component name. You'll use this key to provide a prefix for all the CSS selectors in your component.
```
